data-prepper-plugins/parquet-codecs/src/main/java/org/opensearch/dataprepper/plugins/codec/parquet/DataTypeChecker.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.codec.parquet;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.parquet.example.data.simple.SimpleGroup;
+import org.apache.parquet.schema.PrimitiveType;
+
+import org.apache.parquet.schema.LogicalTypeAnnotation;
+import org.apache.parquet.schema.Type;
+
+import java.util.UUID;
+
+public class DataTypeChecker {
+
+    private DataTypeChecker(){
+
+    }
+
+    public static Object checkDataType(Type field, SimpleGroup simpleGroup, int fieldIndex) throws Exception {
+        String fieldPrimitiveDataType;
+        if (field.isPrimitive()) {
+            PrimitiveType primitiveType = field.asPrimitiveType();
+            fieldPrimitiveDataType = primitiveType.getPrimitiveTypeName().name();
+            switch (fieldPrimitiveDataType) {
+                case "INT32":
+                    return simpleGroup.getInteger(fieldIndex, 0);
+                case "BOOLEAN":
+                    return simpleGroup.getBoolean(fieldIndex, 0);
+                case "FLOAT":
+                    return simpleGroup.getFloat(fieldIndex, 0);
+                case "DOUBLE":
+                    return simpleGroup.getDouble(fieldIndex, 0);
+                case "INT96":
+                    return simpleGroup.getInt96(fieldIndex, 0);
+                case "INT64":
+                case "LONG":
+                    return simpleGroup.getLong(fieldIndex,0);
+                case "FIXED_LEN_BYTE_ARRAY":
+                    return simpleGroup.getValueToString(fieldIndex, 0);
+                default:
+                    throw new RuntimeException("The Parquet Codec doesn't support this data-type yet.");
+            }
+        }
+        else if (field.getLogicalTypeAnnotation() != null) {
+                if (field.getLogicalTypeAnnotation().equals(LogicalTypeAnnotation.dateType())) {
+
+                    return simpleGroup.getInteger(fieldIndex, 0);
+                }
+                else if (field.getLogicalTypeAnnotation().equals(LogicalTypeAnnotation.timestampType(true, LogicalTypeAnnotation.TimeUnit.MICROS))) {
+
+                    return simpleGroup.getLong(fieldIndex, 0);
```

Comment:
It may be good to convert this into a Java-time `Instant` object.

Reply:
Parquet documentation says that timestamp logical datatype has to be expressed as a long. As int96 to be more precise.